magicbot/magicrobot.py

```diff
@@ -433,31 +433,57 @@ def _setup_vars(self, cname, component):
         
         component_type = type(component)
         
+        # Iterate over variables with type annotations
+        for n, inject_type in getattr(component, '__annotations__', {}).items():
+            attr = getattr(component_type, n, True)
```

Comment:
This should probably be moved into the below if statement, since it's not used outside it. You can then remove the default parameter.